src/movies.js

```diff
@@ -1,26 +1,80 @@
 // The `movies` array from the file `src/data.js`.
-console.log('movies: ', movies);
-
+// console.log('movies: ', movies);
 
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+
+function getAllDirectors(movies) {
+  const directors = movies.map(item => {
+    return item.director;
+  });
+  return directors;
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(movies) {
+  const times = movies.filter(item => item.director === 'Steven Spielberg');
+return times.length;
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(movies) {
+  if(movies.length === 0){
+    return 0;
+  }
+
+  movies.forEach(item => {
+    if (typeof item.score != 'number'){
+      item.score = 0;
+    };
+  });
+
+  if (movies.score){
+    movies.score = 0;
+  }
```

Comment:
in your function `movies` is an array, so lines 33-35 makes no sense, cause array of movies does not have property `score`.. Only objects have properties

You can just remove those lines - makes no difference to the code

Reply:
I see the mistake ;)